cannon/mipsevm/tests/fuzz_evm_common64_test.go

```diff
@@ -1,133 +1,98 @@
 package tests
 
 import (
-	"os"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm/exec"
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm/multithreaded"
 	mtutil "github.com/ethereum-optimism/optimism/cannon/mipsevm/multithreaded/testutil"
 	"github.com/ethereum-optimism/optimism/cannon/mipsevm/testutil"
 )
 
-func FuzzStateConsistencyMulOp(f *testing.F) {
-	f.Add(int64(0x80_00_00_00), int64(0x80_00_00_00), int64(1))
-	f.Add(
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_11_22_33_44)),
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_11_22_33_44)),
-		int64(1),
-	)
-	f.Add(
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_80_00_00_00)),
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_80_00_00_00)),
-		int64(1),
-	)
-	f.Add(
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_FF_FF_FF_FF)),
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_FF_FF_FF_FF)),
-		int64(1),
-	)
-
+func FuzzMulOp(f *testing.F) {
 	const opcode uint32 = 28
 	const mulFunct uint32 = 0x2
-	versions := GetMipsVersionTestCases(f)
-	f.Fuzz(func(t *testing.T, rs int64, rt int64, seed int64) {
-		for _, v := range versions {
-			t.Run(v.Name, func(t *testing.T) {
-				mulOpConsistencyCheck(t, versions, opcode, true, mulFunct, Word(rs), Word(rt), seed)
-			})
-		}
-	})
+	multiplier := func(rs, rt Word) uint64 {
+		return uint64(int64(int32(rs)) * int64(int32(rt)))
+	}
+	mulOpCheck(f, multiplier, opcode, true, mulFunct)
 }
 
-func FuzzStateConsistencyMultOp(f *testing.F) {
-	f.Add(int64(0x80_00_00_00), int64(0x80_00_00_00), int64(1))
-	f.Add(
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_11_22_33_44)),
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_11_22_33_44)),
-		int64(1),
-	)
-	f.Add(
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_80_00_00_00)),
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_80_00_00_00)),
-		int64(1),
-	)
-	f.Add(
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_FF_FF_FF_FF)),
-		testutil.ToSignedInteger(uint64(0xFF_FF_FF_FF_FF_FF_FF_FF)),
-		int64(1),
-	)
-
+func FuzzMultOp(f *testing.F) {
 	const multFunct uint32 = 0x18
-	versions := GetMipsVersionTestCases(f)
-	f.Fuzz(func(t *testing.T, rs int64, rt int64, seed int64) {
-		mulOpConsistencyCheck(t, versions, 0, false, multFunct, Word(rs), Word(rt), seed)
-	})
+	multiplier := func(rs, rt Word) uint64 {
+		return uint64(int64(int32(rs)) * int64(int32(rt)))
+	}
+	mulOpCheck(f, multiplier, 0, false, multFunct)
 }
 
-func FuzzStateConsistencyMultuOp(f *testing.F) {
-	f.Add(uint64(0x80_00_00_00), uint64(0x80_00_00_00), int64(1))
+func FuzzMultuOp(f *testing.F) {
+	const multuFunct uint32 = 0x19
+	multiplier := func(rs, rt Word) uint64 {
+		return uint64(uint32(rs)) * uint64(uint32(rt))
+	}
+	mulOpCheck(f, multiplier, 0, false, multuFunct)
+}
+
+type multiplierFn func(rs, rt Word) uint64
+
+func mulOpCheck(f *testing.F, multiplier multiplierFn, opcode uint32, expectRdReg bool, funct uint32) {
```

Comment:
Moved the explicit test case setup (`f.Add(...)`) into this helper - all the values were identical across the tests except for the types (uint64 vs int64).  Set up the fuzzer to generate random `int64` values for all, then cast them to `Word` (`uint64`).